Add windows compatibility

jakob-grabner opened the pull request:
Make swarmer work on windows. 
Fixes #37

simonschiller committed:
Add option to keep existing avds

rtjag committed:
Use "emulator -list-avds" to check for existing emulators

rtjag committed:
Fix exception if swarmer is called with 0 args

rtjag committed:
Add windows compatibility

rtjag committed:
Fix emulator start command

rtjag committed:
Fix tests

rtjag committed:
Revert changes.

rtjag committed:
Revert changes.

artem-zinnatullin commented on swarmer/src/main/kotlin/com/gojuno/swarmer/Emulators.kt:
```diff
@@ -1,27 +1,41 @@
+}
+val emulator = when (os()) {
+    Os.Linux, Os.Mac -> "$androidHome/emulator/emulator"
+    Os.Windows -> "$androidHome/emulator/emulator.exe"
```
`.exe`! RUN DON'T LOOK BACK

artem-zinnatullin commented on swarmer/src/main/kotlin/com/gojuno/swarmer/Emulators.kt:
```diff
@@ -102,8 +116,7 @@ private fun startEmulator(
                 .doOnNext { log("Ports for emulator ${args.emulatorName}: ${it.first}, ${it.second}.") }
                 .flatMap { ports ->
                     startEmulatorProcess(
-                            // Unix only, PR welcome.
```
I knew someone will delete this line haha

artem-zinnatullin commented on swarmer/src/main/kotlin/com/gojuno/swarmer/Emulators.kt:
```diff
@@ -334,3 +347,17 @@ private fun outputDirectory(args: Commands.Start) =
 private fun connectedEmulators(): Single<Set<AdbDevice>> =
         connectedAdbDevices().take(1).toSingle().map { it.filter { it.isEmulator }.toSet() }
+
+private fun os(): Os {
```
We can expose this from Commander https://github.com/gojuno/commander/blob/05d1714151ae98169fcfd300bf55ffc2f93171d0/os/src/main/kotlin/com/gojuno/commander/os/Processes.kt#L129

yunikkk replied:
Lets do it, sounds good

yunikkk replied:
https://github.com/gojuno/commander/pull/23

artem-zinnatullin reviewed (approved):
LGTM

artem-zinnatullin commented:
@yunikkk can you PTAL?

yunikkk commented:
@jakob-grabner 
I'll expose `os()` from commander now, lets use it from there. 
Then LGTM and thanks for contribution!

rtjag committed:
Use `os()` from commander

Update commander to 0.1.8 -> 0.1.10
Update jCommander to 1.71 -> 1.72
Update kotlin to 1.2.41 -> 1.2.61

rtjag committed:
Merge remote-tracking branch 'gojuno/master' into feature/windows_compatibility

jakob-grabner commented:
@yunikkk thanks for exposing `os()` from commander. I updated the PR to use it.
